src/dispatch/DataPlaneClient.ts

```diff
@@ -114,6 +114,11 @@ export class DataPlaneClient {
         const options = {
             method: METHOD,
             protocol: this.config.endpoint.protocol,
+            port:
+                !this.config.endpoint.port ||
```

Comment:
Yup correct! I was doing a null check but of course parseInt will return NaN in that case so its not required. In fact I will simply use Number instead of parseInt because URL class already does the checks for us and we should just rely on it to do the right thing.